SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -0,0 +1,266 @@
+	AddStep(TEXT("SpatialTestRepNotifyAllClientsModifyReplicatedVariables"), FWorkerDefinition::AllClients, [this]() -> bool
+		{
+			// Extra check to ensure that the Old values are correct before changing the replicated variables.
+			if (OldTestInt3 == -3 && OldTestInt4 == -4)
+			{
+				return true;
+			}
+
+			return false;
+		}, [this]()
+		{
+			// Note that TestInt3 is specifically not modified, this will be relevant in the SpatialTestRepNotifyAllClientsCheckValuesAndRepNotifies step.
+
+			TestInt1 = 10;
+			TestInt2 = 20;
+			TestInt4 = 50;
+
+			FinishStep();
+		}, nullptr, 5.0f);
+
+	// The Server modifies the replicated variables once again.
+	AddStep(TEXT("SpatialTestRepNotifyServerChangeReplicatedVariables"), FWorkerDefinition::Server(1), nullptr, [this]()
+		{
+			TestInt1 = 10;
+			TestInt2 = 20;
+			TestInt3 = 30;
+			TestInt4 = 40;
+
+			TestArray.Add(30);
+			FinishStep();
+		});
+
+	// All clients check that the replicated variables were received correctly and that RepNotify acted as expected.
+	AddStep(TEXT("SpatialTestRepNotifyAllClientsCheckValuesAndRepNotifies"), FWorkerDefinition::AllClients, [this]()->bool
+		{
+			// First make sure that we have correctly received the replicated variables, before checking the RepNotify behaviour.
+			if (!(TestInt1 == 10 && TestInt2 == 20 && TestInt3 == 30 && TestInt4 == 40))
+			{
+				return false;
+			}
+
+			if (!(TestArray.Num() == 3 && TestArray[0] == 1 && TestArray[1] == 2 && TestArray[2] == 30))
+			{
+				return false;
+			}
+
+			return true;
+		}, [this]()
+		{
+			// At this point, we have correctly received the values of all replicated variables, therefore the RepNotify behaviour can be checked.
+
+			// Since the RepNotify for this variable is using the default REPNOTIFY_OnChanged, we expect it not to get called on the clients.
+			if (bOnRepTestInt1Called)
+			{
+				FinishTest(EFunctionalTestResult::Failed, TEXT("OnRepTestInt1 should not be called on the clients"));
+				return;
+			}
+
+			// In this case, the RepNotify uses REPNOTIFY_Always so we expect it to be called on the clients.
+			if (!bOnRepTestInt2Called)
+			{
+				FinishTest(EFunctionalTestResult::Failed, TEXT("OnRepTestInt1 should be called on the clients"));
+				return;
+			}
+
+			// From the Clients, we have not modified the value of this particular variable, so we still expect the old value to be the one initially set by the Server in the first step.
+			if (OldTestInt3 != 3)
+			{
+				FinishTest(EFunctionalTestResult::Failed, TEXT("OnRepTestInt3 should have been called with the old value of 3"));
+				return;
+			}
+
+			// Since we have modified this value from the Clients, we expect its value to be the same as set in SpatialTestRepNotifyAllClientsModifyReplicatedVariables.
+			if (OldTestInt4 != 50)
+			{
+				FinishTest(EFunctionalTestResult::Failed, TEXT("OnRepTestInt4 should have been called with the old value of 40"));
+				return;
+			}
+
+			// We consciously differ from native UE here
```

Comment:
Clarifying the difference here would be useful for context

Reply:
Will do, there are 2 places where this should be mentioned, thank you!